### PR TITLE
chore: move protocols `CreateOptions` into interfaces and

### PR DESCRIPTION
**Pull request by danisharora099**
ref: https://github.com/waku-org/js-waku/issues/1000

### CODE REVIEW

**Review by weboko — APPROVED** [review]
cool 😎

File: packages/core/src/lib/filter/index.ts [hunk 1]
```diff
@@ -66,18 +55,21 @@ export type UnsubscribeFunction = () => Promise<void>;
  */
 class Filter implements IFilter {
   multicodec: string;
-  pubSubTopic: string;
+  options: ProtocolCreateOptions;
```
Comment (fryorcraken) [inline packages/core/src/lib/filter/index.ts:58]: Changelog update needed to highlight breaking change in API.

File: packages/core/src/lib/light_push/index.ts [hunk 2]
```diff
@@ -36,40 +37,31 @@ export interface LightPushComponents {
...
-  pubSubTopic?: string;
-}
-
 /**
  * Implements the [Waku v2 Light Push protocol](https://rfc.vac.dev/spec/19/).
  */
 class LightPush implements ILightPush {
   multicodec: string;
-  pubSubTopic: string;
+  options: ProtocolCreateOptions;
```
Comment (fryorcraken) [inline packages/core/src/lib/light_push/index.ts:45]: update changelog to highlight api breaking change

File: packages/core/src/lib/relay/index.ts [hunk 3]
```diff
@@ -73,12 +59,13 @@ class Relay extends GossipSub implements IRelay {
...
       fallbackToFloodsub: false,
     });
+
     super(components, options);
     this.multicodecs = constants.RelayCodecs;
 
     this.observers = new Map();
 
-    this.pubSubTopic = options?.pubSubTopic ?? DefaultPubSubTopic;
+    this.options = options ?? {};
```
Comment (fryorcraken) [inline packages/core/src/lib/relay/index.ts:68]: you end up applyuing the default pubsub topic and every usage of `this.option`. This is cumbersome and prone to developer error.

Apply the default values in the constructor:
```
this.options = Object.assign({pubSubTopic: DefaultPubSubTopic}, options, {})
```

Or same logic in a getter `get options(): ProtocolOptions`

Or create a class that handles the `ProtocolOptions` and automatically apply default values

File: packages/create/src/index.ts [hunk 4]
```diff
@@ -88,7 +88,8 @@ export async function createLightNode(
   const store = wakuStore(options);
   const lightPush = wakuLightPush(options);
   const filter = wakuFilter(options);
-  const peerExchange = wakuPeerExchange(options);
+
```
Comment (fryorcraken) [inline packages/create/src/index.ts:91]: odd

File: packages/interfaces/src/protocols.ts [hunk 5]
```diff
@@ -17,8 +17,26 @@ export interface PointToPointProtocol {
...
+   * The usage of the default pubsub topic is recommended.
+   * See [Waku v2 Topic Usage Recommendations](https://rfc.vac.dev/spec/23/) for details.
+   *
+   */
   pubSubTopic?: string;
+};
+
+//TODO
+// we can probably move `peerId` into `ProtocolCreateOptions` and remove `ProtocolOptions` and pass it in the constructor
+// however, filter protocol can use multiple peers, so we need to think about this
```
Comment (fryorcraken) [inline packages/interfaces/src/protocols.ts:38]: All protocol can use multiple peers. I don't think `PeerId` belongs to `CreateOptions`.

Do note that it makes sense to have `PubSubTopic` in `ProtocolOptions` (too). Let's see how Waku scaling goes:
- https://rfc.vac.dev/spec/51/
- https://rfc.vac.dev/spec/52/

For now I'd remove this todo and leave it as it is and we can decide later once we ahve more information on sharding and Waku usage.

File: packages/create/src/index.ts [hunk 6]
```diff
@@ -88,7 +88,8 @@ export async function createLightNode(
   const store = wakuStore(options);
```
Comment (fryorcraken) [inline packages/create/src/index.ts:88]: You have not removed the `CreateOptions` definition in this file. I expect it to be replaced with `CreateProtocolOPtions` from @waku/interfaces

**Review by fryorcraken — COMMENTED** [review]
Some remaining work to tackle.

**Pull request merged**